applications/Gluon_decentralized_hardware_crypto_wallet_services.md

```diff
@@ -0,0 +1,370 @@
+- Gluon manages Schnorr Multisig Keys that are randomly distributed and encrypted by hardware-secure-modules (HSM). We call them TEA nodes.
+- Users only need to submit a transaction to Gluon, and it will take over the signing and committing to blockchains.
+- Leverages popular biometric technologies in mobile devices to attain better user experiences without compromising security.
+
+Gluon is built into a WebAssembly module loaded and ran inside the T-rust framework (they are both TEA projects). Gluon is an application (we call TeaLeaf); T-rust is a framework (TeaTree). T-rust contains layer1 and layer2; Layer1 is a blockchain built using Substrate, while Layer2 is an exceptionally designed mini OS along with WebAssembly modules that run on wasm runtime powered by [WaSCC](https://wascc.dev). Gluon also contains layer1 and layer2. Layer1 is a Substrate pellet running inside Tea Layer1 blockchain, and layer2 is a set of WebAssembly actors running in every TEA node's TeaRuntime. 
+
+Gluon serves all Polkadot ecosystem dApps and other blockchains as an entry point (portal); it is transparent to other client chains. There is no integration needed. 
+
+We designed and developed Gluon to:
+- Lower the barrier to blockchain end-users; improve user experiences.
+- Protect users' crypto assets.
+- Demonstrate the potential of the Substrate framework and TEA project.
+
+>Please provide the following:
+>  * A brief description of the project.
+>  * An indication of how you will integrate this project into Substrate / Polkadot / Kusama.
+>  * An indication of why your team is interested in creating this project.
+
+### Project Details 
+
+#### Unique features of Gluon
+
+From a users point of view, Gluon works just like a hardware wallet, but with a few differences:
+- A traditional hardware wallet is just one single hardware secure module (HSM), but Gluon is a group of potentially thousands of HSM running consensus together to protect clients' assets. (Every TEA Mining Node is an HSM which can be considered as a hardware wallet)
+- Users need both a Polkadot extension and a Gluon mobile app to access their Gluon account as 2FA. 
+- Users can transfer authentication factors from one device to another without backing up and restoring mnemonic phases.
+- If users lose both their computer (web auth) and phone (mobile auth), social recovery can still protect and recover their new accounts' crypto assets. 
+- Users only need to submit a transaction then Gluon will submit the signature to the client blockchain without further user interaction.
+
+#### Key distribution and protection
+
+![p1 p2 p3](https://raw.githubusercontent.com/tearust/gluon-docs/d5526500c3a27deaa36fc0c0e4a2f2b8e18b1c60/res/Gluonwallet_p123.svg)
+
+By default, Gluon generates 2 out of 3 MultiSig addresses on Polkadot on behalf of users. The three private keys (secret) are P1, P2, and P3. 
```

Comment:
They do not have to be keys from MultiSig pallet. If the native Schnorrkel multisig is merged into the master, we are happy to use native implementation. This is a replaceable module that can switch to native multisig when it is done, or use the pallet if it is not ready. But looks like the review process is delayed so that we can catch up with the native multisig implementation then. Do you want me to change it in the application to native Schnorrkel multisig?

Reply:
yes please - I think that makes more sense given the grant's current requirements.

Reply:
If schnorrkel is not ready and the project is left stalling, we can revert to MultiSig pallet.

Reply:
Done. I have updated the algorithm to use Schnorrkel.
Thanks